runtime/src/main/java/org/corfudb/runtime/collections/CorfuQueue.java

```diff
@@ -382,9 +384,12 @@ public boolean equals(Object o){
     public String toString(){
         StringBuilder stringBuilder = new StringBuilder(corfuTable.size());
         stringBuilder.append("{");
-        for (Map.Entry<CorfuRecordId, ByteString> entry : corfuTable.entrySet()) {
+
+        for (Iterator<Map.Entry<CorfuRecordId, ByteString>> it = corfuTable.entryStream().iterator(); it.hasNext(); ) {
+            Map.Entry<CorfuRecordId, ByteString> entry = it.next();
```

Comment:
This might turn out to be expensive, does this dump the binary payload? I dont think we should do that for logging purposes.